packages/spanner/src/database.js

```diff
@@ -233,6 +244,12 @@ Database.formatName_ = function(instanceName, name) {
  */
 Database.prototype.close = function(callback) {
   var self = this;
+  var leaks = this.pool_.getSessionLeaks();
+
+  if (leaks.length) {
+    callback(new SessionLeakError(leaks));
+    return;
```

Comment:
I initially went that route, but I don't think the pooling library allows itself to be drained until all sessions have released. In my testing it would just hang indefinitely. 

Between this and #2388 I think we may want to look into using a different pooling library. WDYT?